x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -366,6 +371,10 @@ public boolean isAuthenticatedAsApiKey() {
         return authenticatingSubject.getType() == Subject.Type.API_KEY;
     }
 
+    public boolean isAuthenticatedAsRemoteAccess() {
+        return authenticatingSubject.getType() == Subject.Type.REMOTE_ACCESS;
+    }
```

Comment:
Nit: This method feels a bit odd to me. I understand it is modelled after existing methods but it just feels a bit off. I think the remote access authentication object will never have run-as. The authentication from QC can have run-as. We can even decide to support remote access API key run-as in future. But once these two authentication objects are combined, there should never be any possibility of run-as for the final authentication object. This means the method is redundant since we already have `isRemoteAccess` which reads better. I think a more suitable name is probably `isAuthenticatedWithRemoteAccessApiKey`. But now the semantic is different and requires a different implementation. Overall, I'd just drop this method if not absolutely necessary.

Reply:
Agreed, I'm all good to delete with method.